app/shared/controllers/validators/resolvers/ResolveDateRange.scala

```diff
@@ -16,60 +16,56 @@
 
 package shared.controllers.validators.resolvers
 
-import shared.models.domain.DateRange
-import shared.models.errors.{EndDateFormatError, MtdError, RuleEndBeforeStartDateError, StartDateFormatError}
 import cats.data.Validated
 import cats.data.Validated.{Invalid, Valid}
 import cats.implicits._
+import shared.models.domain.DateRange
+import shared.models.errors.{EndDateFormatError, MtdError, RuleEndBeforeStartDateError, StartDateFormatError}
 
 import java.time.LocalDate
+import scala.math.Ordering.Implicits.infixOrderingOps
 
-private[resolvers] class ResolveDateRange private (yearLimits: Option[YearLimits]) extends Resolver[(String, String), DateRange] {
+case class ResolveDateRange(startDateFormatError: MtdError = StartDateFormatError,
+                            endDateFormatError: MtdError = EndDateFormatError,
+                            endBeforeStartDateError: MtdError = RuleEndBeforeStartDateError)
+    extends ResolverSupport {
 
-  def apply(value: (String, String), notUsedError: Option[MtdError], path: Option[String]): Validated[Seq[MtdError], DateRange] = {
-    val (startDate, endDate) = value
+  private def resolveDateRange(parsedStartDate: LocalDate, parsedEndDate: LocalDate): Validated[Seq[MtdError], DateRange] =
```

Comment:
as a private def, perhaps could be stuffed down to the end of the class? Then "apply" could be at the top, easier for someone checking the resolver to see the public interface at the top (I suppose)